src/participant.rs

```diff
@@ -311,10 +311,18 @@ pub(crate) trait InnerProtocolParticipant: ProtocolParticipant {
         T::Value: Default,
     {
         let pid = self.id();
-        if self.local_storage_mut().retrieve_mut::<T>(pid).is_err() {
+        if !self.local_storage_mut().contains::<T>(pid) {
             self.local_storage_mut().store::<T>(pid, Default::default());
         }
-        self.local_storage_mut().retrieve_mut::<T>(pid)
+        self.local_storage_mut()
+            .retrieve_mut::<T>(pid)
+            .ok_or_else(|| {
+                error!(
+                    "Failed to retrieve {} mutably from storage",
+                    std::any::type_name::<T::Value>()
+                );
+                InternalError::InternalInvariantFailed
```

Comment:
suggestion: Maybe clarify why this is an IIV -- like
> Failed to retrieve {} mutably from storage, even though we just added it if it didn't exist already

Not really necessary though.